src/wp-includes/script-loader.php

```diff
@@ -3027,3 +3027,522 @@ function wp_enqueue_block_style( $block_name, $args ) {
+			str_contains( $webfont['font-family'], ' ' ) &&
+			! str_contains( $webfont['font-family'], '"' ) &&
+			! str_contains( $webfont['font-family'], "'" )
+		) {
+			$webfont['font-family'] = '"' . $webfont['font-family'] . '"';
+		}
+
+		foreach ( $webfont as $key => $value ) {
+			/*
+			 * Skip "provider", since it's for internal API use,
+			 * and not a valid CSS property.
+			 */
+			if ( 'provider' === $key ) {
+				continue;
+			}
+
+			// Compile the "src" parameter.
+			if ( 'src' === $key ) {
+				$value = $fn_compile_src( $webfont['font-family'], $value );
+			}
+
+			// If font-variation-settings is an array, convert it to a string.
+			if ( 'font-variation-settings' === $key && is_array( $value ) ) {
+				$value = $fn_compile_variations( $value );
+			}
+
+			if ( ! empty( $value ) ) {
+				$css .= "$key:$value;";
+			}
+		}
+
+		return $css;
+	};
+
+	/**
+	 * Gets the '@font-face' CSS styles for locally-hosted font files.
+	 *
+	 * @since 6.0.0
+	 *
+	 * @uses $registered_webfonts To access and update the registered webfonts registry (passed by reference).
+	 * @uses $fn_order_src To run the function that orders the src.
+	 * @uses $fn_build_font_face_css To run the function that builds the font-face CSS.
+	 *
+	 * @return string The `@font-face` CSS.
+	 */
+	$fn_get_css = static function() use ( &$registered_webfonts, $fn_order_src, $fn_build_font_face_css ) {
+		$css = '';
+
+		foreach ( $registered_webfonts as $webfont ) {
+			// Order the webfont's `src` items to optimize for browser support.
+			$webfont = $fn_order_src( $webfont );
+
+			// Build the @font-face CSS for this webfont.
+			$css .= '@font-face{' . $fn_build_font_face_css( $webfont ) . '}';
+		}
+
+		return $css;
+	};
+
+	/**
+	 * Generates and enqueues webfonts styles.
+	 *
+	 * @since 6.0.0
+	 *
+	 * @uses $fn_get_css To run the function that gets the CSS.
+	 */
+	$fn_generate_and_enqueue_styles = static function() use ( $fn_get_css ) {
+		// Generate the styles.
+		$styles = $fn_get_css();
+
+		// Bail out if there are no styles to enqueue.
+		if ( '' === $styles ) {
+			return;
+		}
+
+		// Enqueue the stylesheet.
+		wp_register_style( 'wp-webfonts', '' );
+		wp_enqueue_style( 'wp-webfonts' );
+
+		// Add the styles to the stylesheet.
+		wp_add_inline_style( 'wp-webfonts', $styles );
+	};
+
+	/**
+	 * Generates and enqueues editor styles.
+	 *
+	 * @since 6.0.0
+	 *
+	 * @uses $fn_get_css To run the function that gets the CSS.
+	 */
+	$fn_generate_and_enqueue_editor_styles = static function() use ( $fn_get_css ) {
+		// Generate the styles.
+		$styles = $fn_get_css();
+
+		// Bail out if there are no styles to enqueue.
+		if ( '' === $styles ) {
+			return;
+		}
+
+		wp_add_inline_style( 'wp-block-library', $styles );
+	};
+
+	/**
+	 * Adds webfonts mime types.
+	 *
+	 * @since 6.0.0
+	 *
+	 * @param array $mime_types Array of mime types.
+	 * @return array Mime types with webfonts formats.
+	 */
+	$fn_add_mime_types = static function( $mime_types ) {
```

Comment:
‼️ I was able to upload a `woff2` file with unfiltered uploads disabled in my wp-config file:

```php
define( 'DISALLOW_UNFILTERED_HTML', true );
define( 'ALLOW_UNFILTERED_UPLOADS', false );
```

I downloaded a copy of Roberto from Google fonts to do so.

Also, I am seeing different mime types in the downloads from Google fonts:
* ttf: `application/x-font-ttf`
* woff2: `application/octet-stream`

The latter may have been due to the way I downloaded it (`curl .... > woff.woff2`). It was only once I changed the mime type that I was able to upload it.

Reply:
Hey @peterwilsoncc, see my notes here https://github.com/WordPress/wordpress-develop/pull/2622#issuecomment-1108527201. Does removing the web fonts from mime types `upload_mimes` filter resolve this issue for you?

Reply:
In my testing, I'm able reproduce the ability to upload font files to Media library when setting the following constants in `wp-config.php`:

```php
define( 'DISALLOW_UNFILTERED_HTML', false );
define( 'ALLOW_UNFILTERED_UPLOADS', true );
```

I was able to upload font files from TT2's assets folder.

After applying https://github.com/WordPress/wordpress-develop/pull/2622/commits/515afda2c1957c97e03ac1a4a48f2788cd812463, uploading a font file is no longer allowed and the following error notification appears:

<img width="703" alt="Screen Shot 2022-04-25 at 9 30 30 AM" src="https://user-images.githubusercontent.com/7284611/165110913-973955de-c5dd-4fb8-a558-43af17585053.png">

Reply:
>Also, I am seeing different mime types in the downloads from Google fonts:
>* `ttf: application/x-font-ttf`
>* `woff2: application/octet-stream`

I did a comparison on Mac using `file --mime-type [filename]`. Here are the results:

* `woff2` font files:
    * Using TT2's `assets/fonts/` all of the `.woff2` files => `application/octet-stream`
    * Via cUrl `curl -L -o roboto.woff2 https://fonts.gstatic.com/s/roboto/v29/KFOmCnqEu92Fr1Mu5mxKKTU1Kvnz.woff2` => `application/octet-stream`
* `ttf` font files:
    * Directly [downloading Roboto zip from Google Fonts](https://fonts.google.com/specimen/Roboto) and then running `file --mime-type Roboto-Black.ttf` (et al) in terminal => `font/sfnt`
    * Using new TT2's `assets/fonts/dm-sans/`  => `font/sfnt`

Reply:
> Thought: Does this code need to specify the font file mime types for generating `font-face` and allowing the browser to request the file from the server?
> 
> I did a test. After removing the `add_filter` in the stopgap (no font file mime types added), it still works => meaning the `font-face` styles get rendered in `<head>` and the browser `GET`s the files from the server.

In that case, I think it is not needed. 

I'll confirm and push to your PR so any modifications to mime type settings can be considered over a longer period.

Reply:
Seeing the same so have removed all the mime type related code in 66a940a1dbc3f0cfcc006675c0365c867ba14c65

Reply:
Yes, it's safe to remove that part of the code. It was part of the grander webfonts API plan to allow uploading font-files in the media library, and then be able to use them in the API. It's something for the future, and not necessary right now. 👍